pkg/scorecard.go

```diff
@@ -115,11 +144,36 @@ func RunScorecard(ctx context.Context,
 		Date: time.Now(),
 	}
 	resultsCh := make(chan checker.CheckResult)
-	go runEnabledChecks(ctx, repo, &ret.RawResults, checksToRun, repoClient, ossFuzzRepoClient,
+
+	// Set metadata for all checks to use. This is necessary
+	// to create remediations from the probe yaml files.
+	ret.RawResults.Metadata = map[string]string{
+		"repository.host":          repo.Host(),
+		"repository.name":          strings.TrimPrefix(repo.URI(), repo.Host()+"/"),
+		"repository.uri":           repo.URI(),
+		"repository.sha1":          commitSHA,
+		"repository.defaultBranch": defaultBranch,
+	}
+
+	go runEnabledChecks(ctx, repo, &ret.RawResults, checksToRun,
+		repoClient, ossFuzzRepoClient,
 		ciiClient, vulnsClient, resultsCh)
 
 	for result := range resultsCh {
 		ret.Checks = append(ret.Checks, result)
 	}
+
+	// Run the probes.
+	var findings []finding.Finding
+	// TODO(#3049): only run the probes for checks.
+	// NOTE: We will need separate functions to support:
+	// - `--probes X,Y`
+	// - `--check-definitions-file path/to/config.yml
+	// NOTE: we discard the returned error because the errors are
+	// already cotained in the findings and we want to return the findings
+	// to users.
+	// See https://github.com/ossf/scorecard/blob/main/probes/zrunner/runner.go#L34-L45.
+	findings, _ = zrunner.Run(&ret.RawResults, probes.All)
```

Comment:
linter is failing you here for ignoring err, which you should be checking. 

My original feedback meant more along the lines of dont run the probes if you're not on experimental